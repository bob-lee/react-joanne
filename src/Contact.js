import React from 'react'
import './Contact.css'

const Contact = () => {
  const isTouchDevice = window.matchMedia('(pointer:coarse)').matches
  return (
    <div className="contact">
      <div className="content">
        <p>
          Are you interested in having your own portrait or have any enquiry?
        </p>
        <p>
          Please send me an email:
        </p>
        <a href="mailto:joannelee133@gmail.com">
          <i className="fa fa-envelope" aria-hidden="true"></i>
          joannelee133@gmail.com
        </a>
        <p>
          or drop me a call:
        </p>
        <a {...isTouchDevice ? {href: 'tel:+64277564652'} : {}}>
          <i className="fa fa-phone" aria-hidden="true"></i>
          +64 027 756 4652
        </a>
      </div>
    </div>
  )
}

export default Contact