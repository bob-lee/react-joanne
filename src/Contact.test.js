import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Contact from './Contact'

it('Contact renders without crashing', () => {
  isTouchDevice(true)
  const div = document.createElement('div')
  ReactDOM.render(<Contact />, div)
})

it('Contact renders correctly for touch device', () => {
  isTouchDevice(true)
  const tree = renderer
    .create(<Contact />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Contact renders correctly for non-touch device', () => {
  isTouchDevice(false)
  const tree = renderer
    .create(<Contact />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
