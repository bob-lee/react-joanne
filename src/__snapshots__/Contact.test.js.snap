// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly for non-touch device 1`] = `
<div
  className="contact"
>
  <div
    className="content"
  >
    <p>
      Are you interested in having your own portrait or have an enquiry?
    </p>
    <p>
      Feel free to send me an email:
    </p>
    <a
      href="mailto:joannelee133@gmail.com"
    >
      <i
        aria-hidden="true"
        className="fa fa-envelope"
      />
       joannelee133@gmail.com
    </a>
    <p>
      or drop me a call:
    </p>
    <a>
      <i
        aria-hidden="true"
        className="fa fa-phone"
      />
       +64 027 756 4652
    </a>
  </div>
</div>
`;

exports[`Contact renders correctly for touch device 1`] = `
<div
  className="contact"
>
  <div
    className="content"
  >
    <p>
      Are you interested in having your own portrait or have an enquiry?
    </p>
    <p>
      Feel free to send me an email:
    </p>
    <a
      href="mailto:joannelee133@gmail.com"
    >
      <i
        aria-hidden="true"
        className="fa fa-envelope"
      />
       joannelee133@gmail.com
    </a>
    <p>
      or drop me a call:
    </p>
    <a
      href="tel:+64277564652"
    >
      <i
        aria-hidden="true"
        className="fa fa-phone"
      />
       +64 027 756 4652
    </a>
  </div>
</div>
`;
